### ISSUE TITLE
[BUG]:  "modular install max" fails  -->  found architecture 'arm64', required architecture 'x86_64'  M1  MacOS 14.0

### ISSUE THREAD

**Issue by TinkeringEngr** [modularml/max#189]
### Bug description

output from command: 
```
modular install max
# Found release for https://packages.modular.com/max @ 24.4.0
# Downloading archive: packages/24.4.0/max-arm64-apple-darwin22.6.0-24.4.0-6-0.tar.zst
Downloaded           [ █████████████████ ] 100% 327MiB/327MiB
# Extracting downloaded archives.
Extracted            [ █████████████████ ] 100% 327MiB/327MiB
# Running post-install scripts...
Mojo Interop         [ █████░░           ]  33%                               [mojo][error] Mojo SDK post-install test "`mojo build test_mandelbrot.mojo`" failed:
SubprocessError: Command '/Users/hack/.modular/pkg/packages.modular.com_max/bin/mojo build /Users/hack/.modular/pkg/packages.modular.com_max/test/test_mandelbrot.mojo -o /tmp/tmph7h43fw6/test_mandelbrot' returned non-zero exit status 1.
  stderr:
    ld: warning: ignoring file '/tmp/mojo_archive-7384d48.a[21](test_mandelbrot.mojo.19.o)': found architecture 'arm64', required architecture 'x86_64'
    ld: warning: ignoring file '/tmp/mojo_archive-7384d48.a[20](test_mandelbrot.mojo.18.o)': found architecture 'arm64', required architecture 'x86_64'
```

   	... with another ~1000 lines indicating found architecture 'arm64', required architecture 'x86_64'

Step 5 fails with:
```
MAX_PATH=$(modular config max.path) \
  && python3 -m pip install --find-links $MAX_PATH/wheels max-engine
Looking in links: /Users/hack/.modular/pkg/packages.modular.com_max/wheels
ERROR: Could not find a version that satisfies the requirement max-engine (from versions: none)
ERROR: No matching distribution found for max-engine
```

$MAX_PATH directory is valid: /Users/hack/.modular/pkg/packages.modular.com_max/

### Steps to reproduce

Following these instructions to install max engine: https://docs.modular.com/max/install

Fails on step 4: modular install max

### System information

```shell
modular 0.8.0 (39a426b5)
Python 3.11.9 (conda create -n max python=3.11)
conda 4.9.2
Homebrew 4.3.8
Mac OSX 14.0 Sonoma
Command Line Tools for Xcode: 15.3
Apple M1
```

**Comment by ehsanmok** [comment]
Thanks for reporting! can you `modular clean` and install again? also please make sure you activate your conda env and use the python >=3.8, < 3.12 for now.

**Comment by TinkeringEngr** [comment]
The problem persists. Even tried creating a new conda environment running python 3.9.

**Comment by TinkeringEngr** [comment]
The problem persists. I even tried creating a new conda environment running python 3.9.

**Comment by jackos** [comment]
Hi this is due to having the x86 version of brew installed, you need to uninstall brew: [https://github.com/modularml/mojo/issues/1078#issuecomment-1771571786](https://github.com/modularml/mojo/issues/1078#issuecomment-1771571786)<br><br>And then install brew again.

**Comment by TinkeringEngr** [comment]
Problem persisted - could be two confounding issues, but likely just conda issue.  Eventually solved using venv. <br> 

```diff 
- x86_64 brew (?)
- atypical conda environment related to eman2 (scientific image processing suite)
```

<br><br>

Uninstalled brew and reinstalled using this .pkg: https://github.com/Homebrew/brew/releases/tag/4.3.9

Output from brew status commands

```
which -a brew
/opt/homebrew/bin/brew
```

```
stat /opt/homebrew/bin/brew
16777239 128748813 -rwxr-xr-x 1 hack admin 0 6722 "Jul  9 15:52:40 2024" "Jul  9 15:52:40 2024" "Jul  9 15:52:40 2024" "Jul  9 15:52:40 2024" 4096 16 0 /opt/homebrew/bin/brew
```

```
brew config
HOMEBREW_VERSION: 4.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1c98a33a376e5b35ebc801d0f7d6ba7b9eb118ac
Last commit: 2 days ago
Core tap JSON: 09 Jul 22:11 UTC
Core cask tap JSON: 09 Jul 22:11 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.UNG03ksd7t/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

```
brew doctor
Your system is ready to brew.
```

```
stat /opt/homebrew/bin/modular
16777239 128832087 lrwxr-xr-x 1 hack admin 0 35 "Jul  9 16:11:28 2024" "Jul  9 16:11:28 2024" "Jul  9 16:11:28 2024" "Jul  9 16:11:28 2024" 4096 0 0 /opt/homebrew/bin/modular
```

```
stat /Users/hack/eman2-sphire-sparx/envs/max3/bin/python
16777239 128839416 lrwxr-xr-x 1 hack staff 0 10 "Jul  9 16:13:02 2024" "Jul  9 16:13:02 2024" "Jul  9 16:13:02 2024" "Jul  9 16:13:02 2024" 4096 0 0 /Users/hack/eman2-sphire-sparx/envs/max3/bin/python
```

```
python --version
Python 3.11.9
```


```diff
- Problem persisted  "found architecture arm64, required architecture x86_64"    ~x1000

```

<br>

## Decided to try venv instead of conda as my conda looked atypical due to eman2  

<br>



```
python3.11 -m venv max-venv2 && source max-venv2/bin/activate
(max-venv2) hack@fly-2 Desktop % which python
/Users/hack/Desktop/max-venv2/bin/python
```


```
modular clean
(max-venv2) hack@fly-2 Desktop % modular install max
# Found release for https://packages.modular.com/max @ 24.4.0
# Downloading archive: packages/24.4.0/max-arm64-apple-darwin22.6.0-24.4.0-6-0.tar.zst
Downloaded           [ █████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████ ] 100% 327MiB/327MiB
# Extracting downloaded archives.
Extracted            [ █████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████ ] 100% 327MiB/327MiB
# Running post-install scripts...
Installed            [ █████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████ ] 100%
                                                                 █
                                                             █  █
                                                       █    ██
                                                       █   ███   █
                                                       ██  ████  █
██      ███████   ████████       █████   ▐███████▌   ████████████
██      ▐█████▌   ███████▌   █   ▐█████▌   █████   ▐██████████████
██   ▌   █████   ▐   ████   ▐█▌   ███████   ▐█▌   █████████████████
██   █   ▐███▌   █   ███▌   ███   ▐███████▌     ▐██████████████████
██   █▌   ███   ▐█   ███   ▐███▌   ████████▌   ▐███████████████████
██   ██   ▐█▌   ██   ██▌           ▐██████▌     ▐██████████████████
██   ██▌   █   ▐██   ██             █████   ▐█▌   █████████████████
██   ███       ███   █▌   ███████   ▐██▌   █████   ▐██████████████
██   ███▌     ▐███   █   ▐███████▌   █   ▐███████▌   ████████████


MAX is now installed! Almost...

## FINISH THE INSTALL

Install the MAX Python package and set environment variables as per:
https://modul.ar/get-started

## NEXT STEPS

You can now access the 'max' and 'mojo' CLI tools.
Enter 'max --help' or 'mojo --help'.

For MAX docs, see https://docs.modular.com.
For MAX code examples, see https://github.com/modularml/max.

(max-venv2) hack@fly-2 Desktop % max
zsh: command not found: max

(max-venv2) hack@fly-2 Desktop % mojo
zsh: command not found: mojo

(max-venv2) hack@fly-2 Desktop % MAX_PATH=$(modular config max.path) \
  && python3 -m pip install --find-links $MAX_PATH/wheels max-engine
Looking in links: /Users/hack/.modular/pkg/packages.modular.com_max/wheels
Processing /Users/hack/.modular/pkg/packages.modular.com_max/wheels/max_engine-24.4.0-cp311-cp311-macosx_11_0_arm64.whl
Collecting numpy>=1.18 (from max-engine)
  Downloading numpy-2.0.0-cp311-cp311-macosx_14_0_arm64.whl.metadata (60 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.9/60.9 kB 313.0 kB/s eta 0:00:00
Processing /Users/hack/.modular/pkg/packages.modular.com_max/wheels/max_engine_libs-24.4.0-py3-none-macosx_13_0_arm64.whl (from max-engine)
Downloading numpy-2.0.0-cp311-cp311-macosx_14_0_arm64.whl (5.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.2/5.2 MB 5.3 MB/s eta 0:00:00
Installing collected packages: numpy, max-engine-libs, max-engine
Successfully installed max-engine-24.4.0 max-engine-libs-24.4.0 numpy-2.0.0

[notice] A new release of pip is available: 24.0 -> 24.1.2
[notice] To update, run: pip install --upgrade pip

(max-venv2) hack@fly-2 Desktop % max
zsh: command not found: max

(max-venv2) hack@fly-2 Desktop % MAX_PATH=$(modular config max.path) \
  && echo 'export MODULAR_HOME="'$HOME'/.modular"' >> ~/.zshrc \
  && echo 'export PATH="'$MAX_PATH'/bin:$PATH"' >> ~/.zshrc \
  && source ~/.zshrc
pyenv:12: command not found: pyenv

(base) (max-venv2) hack@fly-2 Desktop % MAX_PATH=$(modular config max.path) \
  && echo 'export MODULAR_HOME="'$HOME'/.modular"' >> ~/.zshrc \
  && echo 'export PATH="'$MAX_PATH'/bin:$PATH"' >> ~/.zshrc \
  && source ~/.zshrc
zsh: command not found: modular

(base) (max-venv2) hack@fly-2 Desktop % modular
zsh: command not found: modular

(base) (max-venv2) hack@fly-2 Desktop % which modular
modular not found
```

<br>
For whatever reason modular is no longer accessible? Either way this is a new issue unrelated to "found architecture arm64, required architecture x86_64" 
<br><br>

```
(max-venv2) hack@fly-2 Desktop % modular
zsh: command not found: modular

(max-venv2) hack@fly-2 Desktop % echo $PATH
/Users/hack/.modular/pkg/packages.modular.com_max/bin:/Users/hack/.modular/pkg/packages.modular.com_mojo/bin:/Users/hack/.modular/pkg/packages.modular.com_mojo/bin:/Users/hack/homebrew/opt/libpcap/bin:/Library/TeX/texbin:/Users/hack/homebrew/bin:/Users/hack/opt/anaconda3/bin:/Library/Frameworks/Python.framework/Versions/3.9/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/hack/.nexustoolsexport
```

Also, do you guys need a part-time helper? I would love to get involved with Modular long term.  haha

**Comment by jackos** [comment]
Nice good work! We're hiring, can check out the careers page: https://www.modular.com/company/careers
